thoth/common/openshift.py

```diff
@@ -1121,33 +1121,87 @@ def schedule_dependency_monkey(
     def schedule_build_analysis(
         self,
         *,
-        output_image: Optional[str] = None,
         base_image: Optional[str] = None,
-        registry_user: Optional[str] = None,
-        registry_password: Optional[str] = None,
-        registry_verify_tls: bool = True,
-        environment_type: Optional[str] = None,
+        base_image_analysis_id: Optional[str] = None,
+        base_registry_password: Optional[str] = None,
+        base_registry_user: Optional[str] = None,
+        base_registry_verify_tls: bool = True,
+        output_image: Optional[str] = None,
+        output_image_analysis_id: Optional[str] = None,
+        output_registry_password: Optional[str] = None,
+        output_registry_user: Optional[str] = None,
+        output_registry_verify_tls: bool = True,
         buildlog_document_id: Optional[str] = None,
+        buildlog_parser_id: Optional[str] = None,
+        environment_type: Optional[str] = None,
         origin: Optional[str] = None,
+        debug: bool = False,
         job_id: Optional[str] = None,
     ) -> Optional[str]:
         """Schedule a build analysis workflow."""
         if not self.middletier_namespace:
             raise ConfigurationError(
-                "Unable to schedule build report without middletier namespace being set"
+                "Unable to schedule build analysis without middletier namespace being set"
             )
 
-        workflow_id = job_id or self.generate_id("build-analysis")
+        if not base_image and not output_image and not buildlog_document_id:
+            raise ValueError(
+                "Nothing to do, expected any combination of base image, output image and buildlog document id"
+            )
+
+        base_registry_credentials = None
+        if base_registry_user:
+            if not base_registry_password:
+                raise ValueError(
+                    "Base registry user provided but no base registry password supplied"
+                )
+
+            base_registry_credentials = f"{base_registry_user}:{base_registry_password}"
+
+        output_registry_credentials = None
+        if output_registry_user:
+            if not output_registry_password:
+                raise ValueError(
+                    "Output registry user provided but no base registry password supplied"
+                )
+
+            output_registry_credentials = (
+                f"{output_registry_user}:{output_registry_password}"
+            )
+
+        if base_image and not base_image_analysis_id:
+            base_image_analysis_id = self.generate_id("package-extract")
+
+        if output_image and not output_image_analysis_id:
+            output_image_analysis_id = self.generate_id("package-extract")
+
+        if buildlog_document_id and not buildlog_parser_id:
+            buildlog_parser_id = self.generate_id("buildlog-parser")
+
+        if not job_id:
+            job_id = self.generate_id("build-analysis")
```

Comment:
should be buildlogs-analysis? https://github.com/fridex/storages/blob/33f9b76217f17595603f83118658854c871b8651/thoth/storages/buildlogs_analyses.py#L26

Reply:
I wanted to keep this more generic - it's not just build log analysis, but analysis of the whole build - including resulting container images.

Reply:
Thanks for explanation :), then all good for me!